### PR TITLE
fix(rendering): content-contract sliver geometry warns and commits — no more silent viewport freeze

### PR DESCRIPTION
**Pull request by vanyastaff**
## What

Fixes #708 — the permanent viewport freeze at the child-driven paint boundary — and repairs the headless screenshot harness that was hiding every build-during-layout subtree from visual verification. A `sliver` demo triple rides along as the visual proof (and promptly caught #710).

## The #708 fix: content-contract violations warn and commit, matching release Flutter

Root cause (diagnosed with per-child layout instrumentation, then pinned): when a pinned header's seam-built child measures below the header's `layout_extent` (a min-sizing child — a delegate content bug), the header emits `layout_extent > paint_extent`. FLUI's `SliverGeometry::validation_error` treated that as a **fatal** commit-blocker: `validate_layout_output` rejected it, the commit and `clear_needs_layout` were skipped, the parent saw a `SliverGeometry::ZERO` stand-in — and since every retry re-violated, the node's committed geometry stayed frozen at its last pre-boundary value for the rest of the app's life. Silently.

Flutter draws this exact line differently (`sliver.dart:881-894`): `layoutExtent <= paintExtent` and `paintExtent <= maxPaintExtent` are **debug-only asserts**; a release build commits and consumes the geometry as-is (the successor sits at `layout_extent` past the shorter paint). Per FLUI's established policy for content bugs (the viewport's own non-convergence precedent: warn in release, never crash on third-party widget bugs), the two rules move out of `validation_error()` into a new `SliverGeometry::content_contract_violation()`, and `SliverProtocol::validate_layout_output` now **warns and commits** on them. Pipeline-safety rules (NaN/∞/negative) still reject; the layout-poison mechanism for those is untouched (its test fixtures switched to a genuinely-hard violation).

Tests: `a_min_sizing_child_survives_the_child_driven_paint_boundary` (born red at 95.83-stale vs 100; needs the two-pinned-header scene driven by real animation frames at 1s cadence — a root-dirtying `pump` masks the freeze) and `sliver_content_contract_violation_commits_and_stays_clean` (leaf-level pin of commit + clean bit). Both red-verified by restoring the rejection in `validate_layout_output`.

## The screenshot-harness fix: use the shared fixpoint frame

`examples/screenshot.rs` hand-rolled `PipelineOwner::run_frame` — which never services build-during-layout content, so a `SliverAppBar`'s entire delegate subtree captured as empty space. The helper's own doc says neither path may hand-roll this loop; the example now calls the same `BuildOwner::run_frame_with_layout_builders` that `HeadlessBinding::pump_frame` and the live `draw_frame` use.

## The demo triple

`cargo run -p flui --features "material cupertino" --example screenshot -- sliver | sliver-mid | sliver-collapsed` — a pinned `SliverAppBar` + `FlexibleSpaceBar` (title + colored background) over a labeled list, pre-scrolled to 0 / 90 / 500. Verified by eye: expanded 220px bar with the 1.5×-scaled bottom-start title, mid-collapse tracking, pinned 56px toolbar with the background faded exactly at the fade window. The collapsed capture exposed **#710** (SliverAppBar paints no toolbar surface of its own once the flexible background fades) — filed, not fixed here.

## Evidence

- Both #708 pins red→green; the freeze reproduction requires the animation-driven 1s-cadence scene (documented in the test).
- 863 flui-rendering + 3489 flui-objects/widgets/material/cupertino tests green; `just ci` green (log-verified `JUST_CI_EXIT: 0`).
- The three captures inspected visually — the collapse geometry, title scaling, and fade all behave; #710 tracks the one cosmetic gap found.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_019ta5tVUhEAFp17jeadPoxM

### CODE REVIEW

File: crates/flui-rendering/src/constraints/sliver_geometry.rs [hunk 1]
```diff
@@ -380,6 +373,29 @@ impl SliverGeometry {
...
+    /// [`validation_error`](Self::validation_error): a violation here means
+    /// a widget broke its sizing contract (e.g. a persistent header's child
+    /// sized below the header's extent), not that the pipeline cannot
+    /// consume the numbers.
+    ///
+    /// Flutter draws the same line: these two rules are DEBUG-ONLY asserts
+    /// (`sliver.dart:881-894`, `debugAssertIsValid`) — a release build
+    /// commits and consumes the geometry as-is, placing the successor at
+    /// `layout_extent` past the shorter paint. Rejecting them from the
+    /// commit (the pre-#708 behavior) left the node's previous geometry
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/flui-rendering/src/constraints/sliver_geometry.rs]: Partially applied in 4d9f93e1. Repo precedent treats GitHub issue references as durable artifacts (e.g. `issue #556` in shipped flui-testing docs) — unlike the enumerated private planning markers — but the density here was excessive: the invariant is now stated in plain English everywhere, with a single `issue #708` citation kept on the regression test where the reproduction recipe lives.

File: examples/screenshot.rs [hunk 2]
```diff
@@ -98,6 +106,54 @@ fn main() {
...
+                .background(ColoredBox::new(Color::rgb(21, 101, 192))),
+        );
+
+    let mut slivers: Vec<flui_view::BoxedView> = vec![bar.into_view().boxed()];
+    for i in 0..14 {
+        let shade = if i % 2 == 0 { 245 } else { 232 };
+        slivers.push(
+            SliverToBoxAdapter::new()
+                .child(ColoredBox::new(Color::rgb(shade, shade, shade)).child(
+                    SizedBox::new(0.0, 56.0).child(Padding::new(EdgeInsets::all(px(16.0))).child(
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply examples/screenshot.rs]: Fixed in 4d9f93e1 — `SizedBox::height(56.0)` (the demo tree also moved to `examples/sliver_demo.rs`, now a windowed example with live gesture scrolling; the screenshot arms include it).

File: examples/screenshot.rs [hunk 3]
```diff
@@ -98,6 +106,54 @@ fn main() {
     println!("wrote {out_path} ({demo}, {width}x{height})");
 }
 
+/// A Material collapsing-header scene: pinned `SliverAppBar` with a
+/// `FlexibleSpaceBar` (title + gradient-ish colored background) over a list
+/// of labeled rows, mounted pre-scrolled to `offset`.
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply examples/screenshot.rs]: Fixed in 4d9f93e1 — the comment now says what the demo draws.

File: crates/flui-widgets/tests/sliver_persistent_header.rs [hunk 4]
```diff
@@ -372,6 +372,136 @@ fn a_delegate_swap_rebuilds_only_when_should_rebuild_says_so() {
...
+    }
+    laid.tick();
+
+    // pixels = 120 ⇒ remaining = 170 ⇒ the committed paint extent is the
+    // CHILD's 100 — and, critically, layout must still be tracking at all.
+    assert_eq!(
+        laid.sliver_geometry(header).paint_extent,
+        100.0,
+        "the committed geometry must track across the child-driven boundary          (a stale pre-boundary value means the viewport froze — issue #708)"
+    );
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/flui-widgets/tests/sliver_persistent_header.rs:498]: Fixed in 4d9f93e1 — single-line message, no wrapped-string artifact.

File: crates/flui-rendering/tests/sliver_geometry_validation.rs [hunk 5]
```diff
@@ -157,16 +180,49 @@ fn sliver_leaf_layout_rejects_invalid_geometry_before_state_commit() {
...
+    assert_eq!(
+        committed,
+        layout_exceeds_paint_geometry(),
+        "the violating geometry is consumed as-is (Flutter release behavior)"
+    );
+    assert_eq!(
+        entry.state().geometry(),
+        Some(layout_exceeds_paint_geometry()),
+        "the violating geometry must be COMMITTED — leaving the previous          commit in place freezes the node forever (issue #708)"
+    );
```
[the remark this answers is not in the document]
Comment (vanyastaff) [reply crates/flui-rendering/tests/sliver_geometry_validation.rs:209]: Fixed in 4d9f93e1 — same.
